src/Components/Server/src/Circuits/RevalidatingServerAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    /// <summary>
+    /// A base class for <see cref="AuthenticationStateProvider"/> services that receive an
+    /// authentication state from the host environment, and revalidate it at regular intervals.
+    /// </summary>
+    public abstract class RevalidatingServerAuthenticationStateProvider
+        : ServerAuthenticationStateProvider, IDisposable
```

Comment:
In the long term, if SignalR does get a native auth revalidation feature as we are hoping, then this *might* eventually get phased out (marked as `[Obsolete]`; removed in a later major release).

But even if we keep it forever, it doesn't really worry me because it's purely layered on top of concepts that will definitely stick around forever, and it doesn't add any behavior or complexity to more fundamental parts of the system. It's just additive.

Reply:
agreed. It's neither a strange design nor does it have major consequences on anything else.